Formula/libyang.rb

```diff
@@ -0,0 +1,22 @@
+class Libyang < Formula
+  desc "YANG data modelling language parser, toolkit, and API written in C"
+  homepage "https://github.com/CESNET/libyang"
+  url "https://github.com/CESNET/libyang/", :using => :git, :tag => "v1.0-r2", :revision => "347246611b85e05d16f54faaa5697c4b2ee4b468"
+  version "1.0-r2"
+  sha256 "02b0131587cc677cbda1b153f9a9190e03e6143ee90252a68e5500b05e61170b"
+  head "https://github.com/CESNET/libyang/"
+  depends_on "cmake" => :build
+  depends_on "pcre"
+
+  def install
+    mkdir "build" do
+      system "cmake", "..", *std_cmake_args
+      system "make"
+      system "make", "install"
+    end
+  end
+
+  test do
+    system "true"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.